### ISSUE TITLE
OpenAM Configurator failed to create default or custom configuration

### ISSUE THREAD

**Issue by dabkhazi** [OpenIdentityPlatform/OpenAM#57]
OpenAM 14.1.5 (14.1.6-SNAPSHOT) Configurator failed to create default or custom configuration.

My enviroment is Windows7 64bit, Tomcat 8.5.31 or 9.0.8 (i tested on both), Java 1.8.0_171, OpenAM-14.1.5.war or  OpenAM-14.1.6-SNAPSHOT.war (built from source). I tried to make test configurations on my local computer, but all my attempts failed with single error - Unexpected LDAP exception occurred. I tried default configuration and custom with external OpenDJ (4.1.10). In log file i see this error:

06/27/2018 01:48:04:280 PM MSK: Registering service idRepoService.xml
AMSetupServlet.processRequest: errorSMSException Exception Code:5
Message:Unexpected LDAP exception occurred.
--------------------------------------------------
The lower level exception message
Decoding Error: Cannot decode the provided ASN.1 integer element because the length of the element value was not between one and four bytes (actual length was 60)
The lower level exception:
org.forgerock.opendj.ldap.ConnectionException: Decoding Error: Cannot decode the provided ASN.1 integer element because the length of the element value was not between one and four bytes (actual length was 60)
	at org.forgerock.opendj.ldap.LdapException.newLdapException(LdapException.java:153)
	at org.forgerock.opendj.ldap.spi.ResultLdapPromiseImpl.setResultOrError(ResultLdapPromiseImpl.java:132)
	at org.forgerock.opendj.ldap.spi.ResultLdapPromiseImpl.adaptErrorResult(ResultLdapPromiseImpl.java:110)
	at org.forgerock.opendj.grizzly.GrizzlyLDAPConnection.close(GrizzlyLDAPConnection.java:671)
	at org.forgerock.opendj.grizzly.LDAPClientFilter.handleReadException(LDAPClientFilter.java:452)
	at org.forgerock.opendj.grizzly.LDAPClientFilter.handleRead(LDAPClientFilter.java:442)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:539)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:593)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:573)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.forgerock.opendj.ldap.DecodeException: Cannot decode the provided ASN.1 integer element because the length of the element value was not between one and four bytes (actual length was 60)
	at org.forgerock.opendj.ldap.DecodeException.fatalError(DecodeException.java:66)
	at org.forgerock.opendj.grizzly.ASN1BufferReader.readInteger(ASN1BufferReader.java:288)
	at org.forgerock.opendj.io.LDAPReader.readMessage(LDAPReader.java:121)
	at org.forgerock.opendj.grizzly.LDAPClientFilter.handleRead(LDAPClientFilter.java:438)
	... 14 more

I attached full log 
[install.log](https://github.com/OpenIdentityPlatform/OpenAM/files/2141109/install.log)

Please help to understand me what is the problem.

**Comment by kedarjapan** [comment]
Any updates on this issue ? I'm also facing the exact same problem while installing OpenAM server on tomcat. Any pointers/help is much appreciated

**Comment by vharseko** [comment]
cannot reproduce on embedded OpenDJ
please check version external OpenDJ store (DIRECTORY_PORT = 1389
DIRECTORY_SERVER = opendj.example.com
DIRECTORY_SSL = SIMPLE
DS_DIRMGRDN = cn=Directory Manager
DS_DIRMGRPASSWD = #########)

**Comment by vharseko** [comment]
@kedarjapan please attach your logs

**Comment by kedarjapan** [comment]
@vharseko

Thanks for checking. However, I'm facing this issue with default as well as custom configuration.
I've also tried with master branch with embedded tomcat 7 as per given instructions for deployment.
Please assist.

Environment Details:
OS: Windows 7
Tomcat : 8.5.9
JDK : 1.8
External OpenDJ : 2.6.0
OpenAM : 14.1.5

[install.txt](https://github.com/OpenIdentityPlatform/OpenAM/files/2172506/install.txt)

**Comment by vharseko** [comment]
OpenDJ : 2.6.0 will not work, update the environment of MGTS to  [OpenDJ 4](https://github.com/OpenIdentityPlatform/OpenDJ/releases)

**Comment by kedarjapan** [comment]
Thanks @vharseko .
I'll check and let you know if any issues

**Comment by dabkhazi** [comment]
I tested it with OpenDJ 4.0 (Community edition) on my local computer. (p.s. edarjapan is not from mgts)

**Comment by OpenIdentityPlatformCommunity** [comment]
@dabkhazi  please attach logs with embedded OpenDJ

**Comment by ryancogswell** [comment]
I am seeing this same issue upon trying to create the default configuration for OpenAM (latest release 14.1.10).
When I debugged into the point where the exception occurs, I found that the exception occurs twice before the install aborts. At the first exception, the bytes in the buffer when it was trying to read an integer were the bytes for the following string:
`ax="String" type="single_choice" >
          <ChoiceValues>
                <ChoiceValue i18`
For the second exception the bytes in the buffer were:
`
tax="number" type="single">`

In both of these cases, the code was trying to read an integer when the buffer appears to be in the middle of the word "syntax".
I'm running this on Tomcat 9.0.12 using JDK 10.0.2.

[openam_install.log](https://github.com/OpenIdentityPlatform/OpenAM/files/2457924/openam_install.log)
